### ISSUE TITLE
git worktree built on linux used on windows gets upset with .git [see issue #4821]

### ISSUE THREAD

**Issue by anon99** [libgit2/libgit2#4861]
You are opening a _bug report_ against the libgit2 project: we use
GitHub Issues for tracking bug reports and feature requests.  If you
have a question about an API or usage, please ask on StackOverflow:
http://stackoverflow.com/questions/tagged/libgit2.  If you want to
have high-level discussions about the libgit2 project itself, visit
https://github.com/libgit2/discussions.

Otherwise, to report a bug, please fill out the reproduction steps
(below) and delete these introductory paragraphs.  Thanks!

### Reproduction steps
On a linux filesystem which is windows (samba) shared create a repo and a worktree using linux git.
In the windows share in the worktree directory use tortoise git to do something like a log
### Expected behavior
Just works the same as a non-worktree setup ie:- generates the requested log
### Actual behavior
Message "not a git repository" or similar.

Inside the .gitconfig is a Linux path containing /home/username/ etc ending in .git/worktrees/repo-name.

Non worktrees appear to work fine, its this .gitconfig containing a linux (not windows) path that seems to be the offender.

I imagine translating the Linux path to windows as is done elsewhere would fix it, or having a second line in the .gitconfig with the windows path c:\blah\blah and have libgit2 parse that.
### Version of libgit2 (release number or SHA1)
libgit2 0.27.0 libgit2_tgit.dll
### Operating system(s) tested
Windows 10, TortoiseGit 2.7.0.0 git version 2.18.0.windows.1

**Comment by ethomson** [comment]
You haven't really explained what _git_ does here.  What does _libgit2_ do and what does _git_ do?

Can you show some code that fails with libgit2 and the similar commands that work with git?

**Comment by ethomson** [comment]
If git itself begins to support this, we will too.  Until that time, I'm closing this.  Let me know if I misunderstood the issue, please.  Thanks!

**Comment by anon99** [comment]
I played around with this a bit and found cygwin(windows) knows how to flip paths. That left the path origin [w:/blah/blah]. I then found that if $GITDIR is defined using relative paths [../blah/blah], it uses that before the .git contents. Thats close. I suspect, but havent tried, that manually editing .git to use relative paths may work. I initially got stuck on the "\" vs. "/" and had forgotten about cygwins translations. Thanks for your explanations...
